packages/grafana-ui/src/components/Graph/GraphContextMenu.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useContext } from 'react';
+import { ContextMenu, ContextMenuProps } from '../ContextMenu/ContextMenu';
+import { ThemeContext } from '../../themes';
+import { SeriesIcon } from '../Legend/SeriesIcon';
+import { GraphDimensions } from './GraphTooltip/types';
+import {
+  DateTimeInput,
+  FlotDataPoint,
+  getValueFromDimension,
+  getDisplayProcessor,
+  formattedValueToString,
+  Dimensions,
+} from '@grafana/data';
+import { css } from 'emotion';
+
+export type ContextDimensions<T extends Dimensions = any> = { [key in keyof T]: [number, number | undefined] | null };
+
+export type GraphContextMenuProps = ContextMenuProps & {
+  getContextMenuSource: () => FlotDataPoint | null;
+  formatSourceDate: (date: DateTimeInput, format?: string) => string;
+  dimensions?: GraphDimensions;
+  contextDimensions?: ContextDimensions;
+};
```

Comment:
Adding optional props `dimensions` and `contextDimensions` so we can calculate the value, that is going to be rendered and that is going to be the same value as in Tooltip.